docs/rfcs/XeGPU.md

```diff
@@ -253,23 +253,22 @@ Attributes `L1_hint`, `L2_hint`, and `L3_hint` can be applied to prefetch.
 XeGPU.atomic_rmw reuses the arith dialect attribute, ::mlir::arith::AtomicRMWKindAttr.
 In case that certain Xe GPU target does not support atomic operation for a certain data type, the user needs to convert the matrix to the supported datatype to perform the atomic operation.
 
-alloc_nbarrier allocates named barriers. Named barrier is workgroup level resource, shared by all subgroups.
+create_nbarrier creates a set of named barriers with the specified number. Named barrier is workgroup level resource, shared by all subgroups.
 ```mlir
-  XeGPU.alloc_nbarrier %nbarrier_count: i8
+  XeGPU.create_nbarrier %nbarrier_num: i8
 ```
-`create_nbarrier` assigns a role for a specific named barrier to be producer and/or consumer. The returned nbarrier object holds a description of the specified barrier, which encodes all the barrier information.  It also binds the current thread with the named barrier by holding the returned nbarrier object. Multiple threads may bind to the same nbarrier so that they can sync with each other.
+`init_nbarrier` assigns one named barrier with the specified barrier ID to the current thread. Multiple threads may bind to the same named barrier, and the input specifies the total count of threads. The returned nbarrier object holds a description of the specified barrier, which encodes all the barrier information.  
```

Comment:
I am a little bit lost about the second parameter, it specifies the number of nbarriers created by CreateNbarrierOp, or the number of threads is trying to bind to the nbarrier_id?

Reply:
It is the number of threads trying to bind.  CreateNbarrier is actually just allocating resources. Say it allocate total 4 name barrier for a group of 8 threads, but 2 threads may share 1 named barrier, . 

With your question, I think that I should change the name back from CreateNbarrier to AllocNbarrier.